opsdroid/database/mongo/__init__.py

```diff
@@ -65,9 +66,13 @@ async def put(self, key, data):
         """
         _LOGGER.debug("Putting %s into MongoDB.", key)
         if "_id" in data:
-            await self.database[key].update_one({"_id": data["_id"]}, {"$set": data})
-        else:
-            await self.database[key].insert_one(data)
+            response = await self.database[key].update_one(
+                {"_id": data["_id"]}, {"$set": data}
+            )
+            if response.raw_result["updatedExisting"]:
+                return
+
+        await self.database[key].insert_one(data)
```

Comment:
Wrote some code so make it so that `put` either updates the `_id` field if it exists or adds a new record to the collection.